1-JavaScript/Week4/homework/ex2-mondaysWorth.js

```diff
@@ -29,16 +29,29 @@ const mondayTasks = [
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function computeEarnings(tasksArray, hourlyRate) {
+  let startEarning = 0;
+  const taskEarnings = tasksArray.map((taskEarning) => {
+    const { duration } = taskEarning;
+    taskEarning = (duration * hourlyRate) / 60;
+    return (startEarning += taskEarning);
+  });
+  return `€${taskEarnings[taskEarnings.length - 1].toFixed(2)}`;
```

Comment:
good job! 

for this exercise, it matches better the reduce method instead of map. When we want to return a result of a calculation (total earnings for this case) based on values that we have in a list, reduce is our friend.

As this exercise ask you to use map, you could use both! one for mapping data and the other one for calculating the totals.

```
function computeEarnings(tasksArray, hourlyRate) {
 const tasksDurations = tasksArray.map(task => task.duration);
 const totalEarnings = tasksDurations.reduce((total, duration) => {
    return total + (duration * hourlyRate) / 60;
  }, 0);
  
  return totalEarnings;
}
```